validator/client/src/main/java/tech/pegasys/teku/validator/client/loader/ValidatorLoader.java

```diff
@@ -81,13 +91,34 @@ public synchronized void loadValidators() {
         ownedValidators, validatorProviders, graffitiProvider);
   }
 
-  public synchronized MutableValidatorAddResult loadMutableValidator(
+  public synchronized PostKeyResult loadMutableValidator(
       final KeyStoreData keyStoreData, final String password) {
-    if (mutableValidatorSource.isPresent() && mutableValidatorSource.get().canAddValidator()) {
-      return mutableValidatorSource.get().addValidator(keyStoreData, password);
+    if (!(mutableValidatorSource.isPresent()
+        && mutableValidatorSource.get().canAddValidator()
+        && maybeDataDirLayout.isPresent())) {
```

Comment:
nit: I'd suggest extracting these conditions to a method for readability (e.g `if (canAddValidator`)